src/Workspaces/Core/Portable/Extensions/TypeOrNamespaceUsageInfo.cs

```diff
@@ -0,0 +1,120 @@
+        /// </summary>
+        Base = 0x08,
+
+        /// <summary>
+        /// Represents a reference to a type whose instance is being created.
+        /// For example, 'C' in <code>var x = new C();</code>, where 'C' is a named type.
+        /// </summary>
+        ObjectCreation = 0x10,
+
+        /// <summary>
+        /// Represents a reference to a namespace or type within a using or imports directive.
+        /// For example, <code>using NS;</code> or <code>using static NS.Extensions</code> or <code>using Alias = MyType</code>.
+        /// </summary>
+        Import = 0x20,
+
+        /// <summary>
+        /// Represents a reference to a namespace name in a namespace declaration context.
+        /// For example, 'N1' or <code>namespaces N1.N2 { }</code>.
+        /// </summary>
+        NamespaceDeclaration = 0x40,
+    }
+
+    internal static class TypeOrNamespaceUsageInfoExtensions
+    {
+        public static string ToLocalizableString(this TypeOrNamespaceUsageInfo info)
+        {
+            // We don't support localizing value combinations.
+            Debug.Assert(info.IsSingleBitSet());
+
+            switch (info)
+            {
+                case TypeOrNamespaceUsageInfo.Qualified:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_Qualify;
+
+                case TypeOrNamespaceUsageInfo.TypeArgument:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_TypeArgument;
+
+                case TypeOrNamespaceUsageInfo.TypeConstraint:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_TypeConstraint;
+
+                case TypeOrNamespaceUsageInfo.Base:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_BaseType;
+
+                case TypeOrNamespaceUsageInfo.ObjectCreation:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_Construct;
+
+                case TypeOrNamespaceUsageInfo.Import:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_Import;
+
+                case TypeOrNamespaceUsageInfo.NamespaceDeclaration:
+                    return WorkspacesResources.TypeOrNamespaceUsageInfo_Declare;
+
+                default:
+                    Debug.Fail($"Unhandled value: '{info.ToString()}'");
+                    return info.ToString();
+            }
+        }
+
+        public static bool IsSingleBitSet(this TypeOrNamespaceUsageInfo usageInfo)
+            => usageInfo != TypeOrNamespaceUsageInfo.None && (usageInfo & (usageInfo - 1)) == 0;
+
+        public static ImmutableArray<string> ToLocalizableValues(this TypeOrNamespaceUsageInfo usageInfo)
+        {
+            if (usageInfo == TypeOrNamespaceUsageInfo.None)
+            {
+                return ImmutableArray<string>.Empty;
+            }
+
+            var builder = ArrayBuilder<string>.GetInstance();
+            foreach (TypeOrNamespaceUsageInfo value in Enum.GetValues(typeof(TypeOrNamespaceUsageInfo)))
```

Comment:
isn't Enum.GetValues high cost? I remember it showed in etl trace when called many times.

Reply:
also, building this array multiple time can actually cached?

Reply:
The values are actually cached here: https://github.com/dotnet/roslyn/pull/31682/files#diff-9581fa1a899f7ea06db18fff977638f9R68 and I verified under the debugger that it is computed only once for unique enum value. I have also updated SymbolUsageInfo to now explicitly implement `IEquatable<SymbolUsageInfo>` for clarity.